power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -27,6 +28,28 @@ namespace power_grid_model::meta_data {
 constexpr auto const& as_array(msgpack::object const& obj) { return obj.via.array; }
 constexpr auto const& as_map(msgpack::object const& obj) { return obj.via.map; }
 // NOLINTEND(cppcoreguidelines-pro-type-union-access)
+
+constexpr void handle_inf(msgpack::object const& obj, auto& attr) {
+    using namespace msgpack;
+    using namespace std::string_view_literals;
+
+    auto infinity = std::numeric_limits<double>::infinity();
+    if (obj.type == type::STR) {
+        auto const& obj_string = obj.as<std::string_view>();
+        if (obj_string == "inf" || obj_string == "-inf" || obj_string == "+inf") {
+            if (obj_string == "inf" || obj_string == "+inf") {
+                attr = infinity;
+            } else {
+                attr = -infinity;
+            }
+        } else {
+            throw type_error();
+        }
+    } else {
+        obj >> attr;
```

Comment:
if you use the `return result` approach, you can directly `return infinity` vs `return -infinity` and you don't need as much nesting.

even more so if you reverse the order `if (obj.type != type::STR) { // unpack and return float }` there's no nesting or `else` statements at all anymore

Reply:
I can't really visualize this suggestion, so I didn't choose the return option

Reply:
that would be something of the order of

```cpp
template <std::floating_point Float>
auto unpack(msgpack::object<Stream>& obj) -> Float {
  if (obj.type != msgpack::type::STR) {
    return obj.as<Float>();
  }

  auto const value = obj.as<std::string_view>();
  if (value == "inf"sv || value == "+inf"sv) {
    return infinity;
  }
  if (value == "-inf"sv) {
    return -infinity;
  }

  throw msgpack::type_error();
}

// usage
{
  double attr = unpack<double>(obj);
}
```

Reply:
that way, there's no nested if/else statements and the logic flow.

small sidenote: you unfortunately cannot use `msgpack::type::FLOAT` because there's ambiguity between `msgpack::type::FLOAT`, `msgpack::type::FLOAT32` and `msgpack::type::FLOAT64`. the actual floating point encoding should not affect the floating point value (which may have a different value type)